creating_citation_counts_tsv/scripts/summarize_gene_citations_all_species.py

```diff
@@ -0,0 +1,335 @@
+                    .map(lambda x: (str(x[1]), int(x[0])))
+                    .filter(lambda x: x[0] in pubmeds_set))
+    
+    # Outputing the first element in the year_pmid variable
+    print(year_pmid.take(1))
+    
+    # Getting a list of the gene ID and gene publication citation ID pairs over the past 5 months and saving it to the gene2pubmed_to_year variable
+    gene2pubmed_to_year = gene2pubmed.join(year_pmid).map(lambda x: (x[1][0], x[0]))
+    
+    # Outputing the first element in the gene2pubmed_to_year variable
+    print(gene2pubmed_to_year.take(1))
+    
+    # Getting a list of the gene ID, gene publication citation ID, count of how frequently the gene is cited over the past 5 months and saving it to the gene_counts variable
+    gene_counts = gene2pubmed.join(year_pmid).map(lambda x: (x[1][0], 1)).reduceByKey(add).join(gene2pubmed_to_year).map(lambda x: ( str(x[0]), (x[1][1] ,x[1][0])))
+    
+    # Outputing the first element in the gene_counts variable
+    print(gene_counts.take(1))
+    
+    # Returning the gene_counts variable
+    return gene_counts
+
+def getting_gene_info(species, gene_counts):
+    """ 
+    Getting a list of the gene information per gene and species' taxonomy name.
+    """
+    
+    # Adding gene's information to the list created from getting_recent_gene2pubmed per gene and saving it to the gene_info variable
+    gene_info = (sc.textFile(f"creating_citation_counts_tsv/data/{species}/gene_info")
+                    .filter(lambda x: x[0] != '#')
+                    .map(lambda x: x.split('\t'))
+                    .map(lambda x: (str(x[1]), (str(x[0]), str(x[2]), str(x[6]), str(x[8]))))).join(gene_counts).map(lambda x: (x[1][0][1].upper(), (x[0], x[1][0][2], x[1][0][3], x[1][1][1], x[1][0][0])))
+    
+    # Outputing the first element in the gene_info variable
+    print(gene_info.take(1))
+    
+    # Getting the taxonomy ID for the species and saving it to the tax_id variable
+    tax_id = str(gene_info.collect()[0][1][4])
+    
+    # Outputing the first element in the tax_id variable
+    print(tax_id)
+
+    # Getting the taxonomy information for the taxonomy ID given and saving it to the taxonomy variable
+    taxonomy = (sc.textFile("creating_citation_counts_tsv/taxonomy_name")
+                        .filter(lambda x: x[0] != '#')
+                        .map(lambda x: x.split('\t'))
+                        .filter(lambda x: int(tax_id) == int(x[0]) )
+                        .filter(lambda x: "scientific name" in x[6] )
+                        .map(lambda x: ((str(x[0]), str(x[2]), str(x[6]))))
+                     )
+    
+    # Getting the taxonomy name from the taxonomy information and saving it to the tax_name variable
+    tax_name = taxonomy.collect()[0][1]
+    
+    # Outputing the tax_name variable
+    print(tax_name)
+
+    # Returning the gene_info and tax_name variable
+    return gene_info, tax_name
+
+def getting_refGene(species, gene_info): 
```

Comment:
Python conventionally uses snake_case instead of camelCase.

```suggestion
def get_ref_gene(species, gene_info): 
```

(Either `refgene` or `ref_gene` would be conventional, but not `refGene)